File: game/game_constants.py
from enum import Enum


class GamePhase(Enum):
    PROPOSAL = 0
    VOTE = 1
    MISSION = 2
    ASSASSINATION = 3


class LobbyStatus(Enum):
    JOINING = 0
    IN_PROGRESS = 1
    DONE = 2


class MissionCard(Enum):
    SUCCESS = 0
    FAIL = 1
    REVERSE = 2


class MissionResult(Enum):
    PASS = 0
    FAIL = 1

File: thavalon/typescript/gameConstants.ts
export enum Card {
    Success,
    Fail,
    Reverse
}

export const enum GamePhase {
    Proposal,
    Voting,
    Mission,
    Assassination
}

export const enum Vote {
    Downvote,
    Upvote
}

export const enum Team {
    Good,
    Evil
}

export const enum MissionResult {
    Pass,
    Fail
}

export interface RoleInformation {
    role: string;
    roleSummary: string;
    team: Team;
}

export interface AllMissionInfo {
    missionNumber: number;
    singleMissionInfo: missionInfo;
}

export interface missionInfo {
    missionResult: MissionResult;
    playersOnMission: string[];
    playedCards: Card[];
}

